.circleci/config.yml

```diff
@@ -1,6 +1,23 @@
 version: 2.1
 
 jobs:
+  common-precommit-checks:
+    docker:
+      - image: circleci/python:3.8
+    steps:
+      - checkout
+      - run:
+          name: Install pre-commit
+          command: |
+            pip install pre-commit
+      - run:
+          name: Run pre-commit hooks
+          command: |
+            changed_files=$(git diff --name-only $CIRCLE_BRANCH...HEAD)
```

Comment:
This is copied from .travis.yml, but unfortunately apparently it is broken in subtle ways: see https://github.com/RespiraWorks/VentilatorSoftware/issues/184

I suggest to hold off this PR until that issue is fully fixed.

Reply:
Updated based on a slack conversation with @jlebar 

Now it:
- Finds the common branch between itself and origin/master
- Resets all changed files since then
- Runs precommit against them